shell/platform/android/io/flutter/plugin/mouse/MouseCursorController.java

```diff
@@ -0,0 +1,119 @@
+// Copyright 2019 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.plugin.mouse;
+
+import android.content.Context;
+import android.support.annotation.NonNull;
+import android.view.View;
+import android.view.PointerIcon;
+
+import java.util.HashMap;
+
+import io.flutter.embedding.engine.dart.DartExecutor;
+import io.flutter.embedding.engine.systemchannels.MouseCursorChannel;
+import io.flutter.plugin.mouse.MouseCursors;
+
+/**
+ * Manages mouse cursor of the mouse device.
+ *
+ * Android only allows one pointer device.
+ *
+ * Each cursor has a representation, which is an integer, and has an implementation, which is a
+ * {@code PointerIcon} object. System cursors, which are the constants defined in
+ * {@code MouseCursors}, can be directly used, while you can also use
+ * {@code MouseCursorController.registerCursor} to define custom cursors or overwrite system
+ * cursors. Unregistered cursors will fallback to {@code MouseCursors.basic}.
+ *
+ * This class only communicates in Flutter's cursor constants, which means it doesn't recognize the
+ * constants defined in {@code PointerIcon}.
+ */
+public class MouseCursorController {
+  private static final String TAG = "MouseCursorController";
+
+  /**
+   * Constructs a {@code MouseCursorController} that controls the cursor on the given {@code view}.
+   *
+   * The given {@code dartExecutor} is used to construct the method channel with which the
+   * platform communicates with the framework.
+   *
+   * The given {@code context} is used to create system cursor object. Usually it's the current
+   * activity.
+   */
+  public MouseCursorController(
+      @NonNull View view,
+      @NonNull DartExecutor dartExecutor,
+      @NonNull Context context
+  ) {
+    channel = new MouseCursorChannel(dartExecutor);
+    channel.setMethodHandler(channelHandler);
+    this.view = view;
```

Comment:
Rather than manipulate a `View` within this class, I would recommend defining a delegate interface and then implement that interface within the old and new `FlutterView`s. That way, there is no magic or mystery when reading `FlutterView` as to what it's doing. When we move control of something like a `View` to other classes, sooner or later we get surprised by conflicts or unexpected behaviors.